### PR TITLE
New hallucinations

### PR DESCRIPTION
**Pull request by tigercat2000**
tgstation/-tg-station#9791
>This commit adds a number of new hallucinations and increases the
modularness of the hallucination system. Included hallucinations:
 - Alien Hunter
  - Alien hunter comes out of the nearest vent, leaps and stuns you, then
    crawls back into the vent.
 - Plasma flood
 - Singularity
  - Stage 4 singularity appears and chases you.
 - Fight sounds
  - Series of sounds that sound like a fight.
 - Clown-visions
  - Delusion, which makes an image of a mob standing still.
 - Fake bolted doors
  - A number of doors will appear to become bolted, via overlay.
 - Fake speech/radio messages.
  - This makes it appear as if someone has said something, either via
    speech or radio, using one language from their avalible languages.
    Radio messages will only happen if there are no people in view to say
    something.

Basically, actually kindof scary hallucinations, alongside the old boring ones.

### CODE REVIEW

**Comment by Spacemanspark** [comment]
:+1:

**Comment by Fox-McCloud** [comment]
Was wanting to implement this myself, but you beatttt me.

:+1:

File: code/modules/flufftext/Hallucination.dm [hunk 1]
```diff
@@ -19,187 +19,349 @@ mob/living/carbon/
...
+	var/obj/machinery/atmospherics/unary/vent_pump/pump = null
+	var/obj/effect/hallucination/simple/xeno/xeno = null
+
+/obj/effect/hallucination/xeno_attack/New(loc,var/mob/living/carbon/T)
+	target = T
+	for(var/obj/machinery/atmospherics/unary/vent_pump/U in orange(7,target))
+		if(!U.welded)
+			pump = U
+			break
+	xeno = new(pump.loc,target)
```
Comment (Krausus) [inline code/modules/flufftext/Hallucination.dm]: This will runtime if someone gets this hallucination while they aren't within 7 tiles of a vent. Also, `orange` will pick vents that aren't even in your view, and I have a feeling that always grabbing the first one found will bias it towards a certain direction.

File: code/modules/flufftext/Hallucination.dm [hunk 2]
```diff
@@ -19,187 +19,349 @@ mob/living/carbon/
...
+	sleep(10)
+	xeno.update_icon("alienh_leap",'icons/mob/alienleap.dmi',-32,-32)
+	xeno.throw_at(target,7,1)
+	sleep(10)
+	xeno.update_icon("alienh_leap",'icons/mob/alienleap.dmi',-32,-32)
+	xeno.throw_at(pump,7,1)
```
Comment (Krausus) [inline code/modules/flufftext/Hallucination.dm]: Sleeping without sanity-checking any object references before trying to use them is a runtime waiting to happen. In this case, the odds are extremely low, but it's still a bad practice.

**Comment by Allfd** [comment]
:+1:   This should be fun, especially the first time people not on github encounter the xeno.

**Comment by fj45** [comment]
:100:

**Comment by FalseIncarnate** [comment]
Hallucinating speech? Gonna be trickier for me to just ignore it now!

:+1:

**Comment by Ultimate-Chimera** [comment]
Alium Hallucinations without adminbus now? https://www.youtube.com/watch?v=FrZRIW87eWI

**Comment by monkeysfist101** [comment]
Throw in some radio calls such as, "SECURITY TO THE BAR," "Plasma is leaking in Central," and "X is a changeling," and I'm sold.

**Comment by tigercat2000** [comment]
@monkeysfist101 can add those later, new phrases are easy. Don't really want to touch this PR, since it's a 1:1 port right now.
